### PR TITLE
Added patch for services.internal_port

### PR DESCRIPTION
**Pull request by billyb2**

### CODE REVIEW

File: internal/appv2/patches.go [hunk 1]
```diff
@@ -198,6 +198,23 @@ func _patchService(service map[string]any) (map[string]any, error) {
...
+		switch cast := rawInternalPort.(type) {
+		case string:
+			n, err := strconv.Atoi(cast)
+			if err != nil {
+				return nil, fmt.Errorf("Can not convert internal_port '%s' to integer: %w", cast, err)
+			}
+			service["internal_port"] = n
+		case float64:
+			service["internal_port"] = int(cast)
+		case int64:
```
Comment (tvdfly) [inline internal/appv2/patches.go:211]: I think int16, int32, int, and the uint variations would also all be acceptable to cast to int.

File: internal/appv2/patches.go [hunk 2]
```diff
@@ -198,6 +198,23 @@ func _patchService(service map[string]any) (map[string]any, error) {
...
 
+	if rawInternalPort, ok := service["internal_port"]; ok {
+		switch cast := rawInternalPort.(type) {
+		case string:
+			n, err := strconv.Atoi(cast)
+			if err != nil {
+				return nil, fmt.Errorf("Can not convert internal_port '%s' to integer: %w", cast, err)
+			}
+			service["internal_port"] = n
+		case float64:
```
Comment (tvdfly) [inline internal/appv2/patches.go:209]: how about handling float32 too?

File: internal/appv2/patches.go [hunk 3]
```diff
@@ -198,6 +198,23 @@ func _patchService(service map[string]any) (map[string]any, error) {
...
+			service["internal_port"] = int(cast)
+		case int64:
+			service["internal_port"] = int(cast)
+		default:
+			return nil, fmt.Errorf("Unknown type for internal_port number: %T", cast)
+		}
```
Comment (gwuah) [inline internal/appv2/patches.go:215]: [question/suggestion] would the type still be **int64/float64** if the person was running flyctl on a 32 bit machine? (idk for sure, im just wondering)

What do you think about something like 👇 ?
```suggestion
		switch cast := rawInternalPort.(type) {
		case string:
			n, err := strconv.Atoi(cast)
			if err != nil {
				return nil, fmt.Errorf("Can not convert internal_port '%s' to integer: %w", cast, err)
			}
			service["internal_port"] = n
		default:
		        port, ok := rawInternalPort.(int)
		        if !ok {
		            return nil, fmt.Errorf("Unknown type for internal_port number: %T", cast)
		        }
		        service["internal_port"] = port
		}
```

Comment (billyb2) [reply internal/appv2/patches.go:215]: This solution looks really clean, but doesn't work unfortuanetly. It works great for string ports, but when I try it with a regular int port I get `Error invalid config: Unknown type for internal_port number: int64`

File: internal/appv2/patches.go (hunk 1 above)
Comment (billyb2) [reply internal/appv2/patches.go]: You're right, got them all

File: internal/appv2/patches.go (hunk 2 above)
Comment (billyb2) [reply internal/appv2/patches.go]: Roger

File: internal/appv2/patches.go (hunk 3 above)
Comment (gwuah) [reply internal/appv2/patches.go]: day by day, my reputation as a liar goes up. 🥱

**Comment by tvdfly** [comment]
this failed when I tested with this fly.toml:

```toml
app = "tvd-test-apps-v2"
[[services]]
  internal_port = "80"
  protocol = "tcp"
  [[services.ports]]
    force_https = true
    handlers = ["http"]
    port = 80
  [[services.ports]]
    handlers = ["tls", "http"]
    port = 443

```

```output
% fly version
flyctl v0.0.0-1677043537+dev darwin/arm64 Commit: 9ee38235147acbe35a6786107886ac140f42d60b (internal_command_fix) BuildDate: 2023-02-22T05:25:37Z

% fly deploy
Error failed loading app config from testing-fly.toml: toml: line 3 (last key "services.internal_port"): incompatible types: TOML value has type string; destination has type integer

```

**Comment by billyb2** [comment]
I'm not able to reproduce this on my machine. Do you mind if I ask what OS you're running on?

**Comment by tvdfly** [comment]
np, macos 13 with an m1 chip. let's see if I can repro again and/or in a docker container

**Comment by tvdfly** [comment]
yep, repros for me if I use a golang docker container when building directly off this branch. You will hopefully be able to repro using docker.

I double checked the commit sha shown in version matches the most recent commit here (at the time I ran this).

```
% docker run -it --rm --platform linux/amd64 golang:1.19 /bin/bash
root@2453f0a96a5a:/go# git clone https://github.com/superfly/flyctl --branch internal_command_fix
root@2453f0a96a5a:/go# cd flyctl/
root@2453f0a96a5a:/go/flyctl# make
...

root@2453f0a96a5a:/go/flyctl# ./bin/flyctl version
flyctl v0.0.0-1677139686+dev linux/amd64 Commit: 9ee38235147acbe35a6786107886ac140f42d60b (internal_command_fix) BuildDate: 2023-02-23T08:08:06Z

root@2453f0a96a5a:/go/flyctl# ./bin/flyctl auth login
...

root@2453f0a96a5a:/go/flyctl# cat<<EOF >fly.toml
app = "tvd-test-apps-v2"
[[services]]
  internal_port = "80"
  protocol = "tcp"
  [[services.ports]]
    force_https = true
    handlers = ["http"]
    port = 80
  [[services.ports]]
    handlers = ["tls", "http"]
    port = 443
EOF

root@2453f0a96a5a:/go/flyctl# ./bin/flyctl status
Error failed loading app config from /go/flyctl/fly.toml: toml: line 3 (last key "services.internal_port"): incompatible types: TOML value has type string; destination has type integer

root@2453f0a96a5a:/go/flyctl# ./bin/flyctl deploy
Error failed loading app config from /go/flyctl/fly.toml: toml: line 3 (last key "services.internal_port"): incompatible types: TOML value has type string; destination has type integer
```

**Comment by billyb2** [comment]
This is what happens running on my end (with Docker):
```
root@ddda4c024bd2:/go/flyctl/hello# cat fly.toml
app = "billy-test-apps-v2"
[[services]]
  internal_port = "80"
  protocol = "tcp"
  [[services.ports]]
    force_https = true
    handlers = ["http"]
    port = 80
  [[services.ports]]
    handlers = ["tls", "http"]
    port = 443
root@ddda4c024bd2:/go/flyctl/hello# ../bin/flyctl version
flyctl v0.0.0-1677169326+dev linux/amd64 Commit: 9ee38235147acbe35a6786107886ac140f42d60b (internal_command_fix) BuildDate: 2023-02-23T16:22:06Z
root@ddda4c024bd2:/go/flyctl/hello# ../bin/flyctl status
App
  Name     = billy-test-apps-v2
  Owner    = personal
  Version  = 0
  Status   = pending
  Hostname = billy-test-apps-v2.fly.dev
  Platform =

App has not been deployed yet.
root@ddda4c024bd2:/go/flyctl/hello# ../bin/flyctl deploy --image nginx
==> Verifying app config
--> Verified app config
==> Building image
Searching for image 'nginx' remotely...
image found: img_19gm46k36kzvx0jk
==> Creating release
--> release v2 created

--> You can detach the terminal anytime without stopping the deployment
==> Monitoring deployment
Logs: https://fly.io/apps/billy-test-apps-v2/monitoring

v0 is being deployed

```

Super confused if I'm being honest. I'll see if I can get a third person to test this. It definetely shouldn't be merged until we figure this out

**Comment by billyb2** [comment]
When I'm added to @tvdfly 's test organization, then the patch stops working on both my machine and Docker. Looking into it further...

**Comment by dangra** [comment]
Not sure why I missed this PR's review but I hope we can add a test case at https://github.com/superfly/flyctl/blob/master/internal/appv2/serde_test.go

**Comment by dangra** [comment]
Tested by https://github.com/superfly/flyctl/pull/1827

**Pull request merged**